src/python/tests/conftest.py

```diff
@@ -207,12 +68,12 @@ def test_method(..., dir_cmp: DirCmp, ...):
 
     """
     # Get the source and temporary absolute paths for reference and target root directories
-    ref = Path(request.param['ref'])
-    ref_src = ref if ref.is_absolute() else pytest.files_dir / ref
-    ref_tmp = tmp_dir / str(ref).replace(os.path.sep, '_')
-    target = Path(request.param['target'])
-    target_src = target if target.is_absolute() else pytest.files_dir / target
-    target_tmp = tmp_dir / str(target).replace(os.path.sep, '_')
+    ref = Path(request.param['ref'])  # type: ignore[attr-defined]
+    ref_src = ref if ref.is_absolute() else pytest.files_dir / ref  # type: ignore[attr-defined]
+    ref_tmp = Path(tmp_dir) / str(ref).replace(os.path.sep, '_')
+    target = Path(request.param['target'])  # type: ignore[attr-defined]
+    target_src = target if target.is_absolute() else pytest.files_dir / target  # type: ignore[attr-defined]
+    target_tmp = Path(tmp_dir) / str(target).replace(os.path.sep, '_')
     # Copy directory trees (if they have not been copied already) ignoring file metadata
     if not ref_tmp.exists():
         shutil.copytree(ref_src, ref_tmp, copy_function=shutil.copy)
```

Comment:
Again out of curiosity, why does this use `shutil.copy` instead of the default `shutil.copy2`? Is it related to the need to have different timestamps between reference and target files?

Reply:
Yes, that is required to ensure `filecmp` methods are forced to inspect the content of the file even when the size matches.